src/sinks/nats.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{
+    config::{DataType, GenerateConfig, SinkConfig, SinkContext, SinkDescription},
+    emit,
+    event::Event,
+    internal_events::NatsEventSent,
+    sinks::util::StreamSink,
+};
+use async_trait::async_trait;
+use futures::{future, stream::BoxStream, FutureExt, StreamExt};
+use serde::{Deserialize, Serialize};
+
+/**
+ * Code dealing with the SinkConfig struct.
+ *
+ * DEV: Start with the bare minimum for now.
+ */
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct NatsSinkConfig {
+    url: String,
+    subject: String,
+}
+
+inventory::submit! {
+    SinkDescription::new::<NatsSinkConfig>("nats")
+}
+
+impl GenerateConfig for NatsSinkConfig {}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "nats")]
+impl SinkConfig for NatsSinkConfig {
+    async fn build(
+        &self,
+        _cx: SinkContext,
+    ) -> crate::Result<(super::VectorSink, super::Healthcheck)> {
+        let sink = NatsSink::new(self.clone());
+        let healthcheck = future::ok(()).boxed();
```

Comment:
Could we have this healthcheck just try to connect, similar to the socket sinks?